### PR TITLE
Add experiment args 

### PR DESCRIPTION
**Pull request by sourabhagr**
### Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### Description

Please include a summary of the change and which issue it addresses. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

### Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/uptrain-ai/uptrain/blob/main/CONTRIBUTING.md) document.
- [ ] My code follows the code style (BLACK) of this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated the documentation accordingly.
- [ ] I have added an appropriate CHANGELOG entry.

### Author's Note

Please provide any additional information, questions, or concerns you have about this pull request.

### CODE REVIEW

**Comment by sourabhagr** [comment]
Cleanup needs to be done. Don't merge!

File: uptrain/operators/io/readers.py [hunk 1]
```diff
@@ -27,11 +47,10 @@ class CsvReader(TransformOp):
     batch_size: t.Optional[int] = None
 
     def setup(self, settings: Settings):
-        self._executor = TextReaderExecutor(self)
         return self
 
     def run(self) -> TYPE_TABLE_OUTPUT:
-        return {"output": self._executor.run()}
+        return {"output": TextReaderExecutor(self).run()}
```
Comment (sourabhagr) [inline uptrain/operators/io/readers.py:53]: @ananis25 is it okay to make this change?

File: uptrain/operators/table.py [hunk 2]
```diff
@@ -81,3 +81,67 @@ def run(self, data: pl.DataFrame) -> TYPE_TABLE_OUTPUT:
...
+
+    col_in_names: list[str]
+    col_out: str
+
+    def setup(self, settings: Settings):
+        assert len(self.col_in_names) > 1
+        return self
+
+    def run(self, data: pl.DataFrame) -> TYPE_TABLE_OUTPUT:
+        return {"output": data.with_columns(data.select(
```
Comment (sourabhagr) [inline uptrain/operators/table.py:144]: @ananis25, can we do it in a for loop?

File: uptrain/operators/language/llm.py [hunk 3]
```diff
@@ -70,6 +69,7 @@ async def async_process_payload(
                 ):
                     await asyncio.sleep(3)
                 else:
+                    payload.error = str(exc)
```
Comment (ananis25) [inline uptrain/operators/language/llm.py:72]: nice!

File: uptrain/framework/remote.py [hunk 4]
```diff
@@ -84,6 +84,19 @@ def list_checksets(self, skip: int = 0, limit: int = 10):
         response = self.client.get(url, params=params)
         return response.json()
 
+    def add_experiment(self, name: str, checkset: CheckSet, experiment_args: ExperimentArgs):
```
Comment (ananis25) [inline uptrain/framework/remote.py:87]: This function also needs to take the settings object.

File: uptrain/operators/language/model_grade.py [hunk 5]
```diff
@@ -105,7 +105,7 @@ def setup(self, settings: Settings):
     def _make_payload(self, id: t.Any, messages: list[dict]) -> Payload:
         return Payload(
             endpoint="chat.completions",
-            data={"model": "gpt-3.5-turbo", "messages": messages},
+            data={"model": "gpt-3.5-turbo", "messages": messages, "temperature": 0.2},
```
Comment (ananis25) [inline uptrain/operators/language/model_grade.py:108]: temperature needs to be the variable declared earlier?

File: uptrain/framework/checks.py [hunk 6]
```diff
@@ -225,3 +227,68 @@ def serialize(self, fpath: t.Optional[str] = None):
...
+
+    prompt_template: str
+    prompt_params: dict = {}
+    models: t.Union[str, list[str]]
+    context_vars: t.Union[dict, list]
+    temperature: float = 1.0
+    col_out_response: str = 'response'
+    col_out_mapping: t.Optional[dict] = None
+
+    def get_preprocessors(self):
```
Comment (ananis25) [inline uptrain/framework/checks.py:254]: if users dont use these methods, can we add an underscore prefix?

**Review by ananis25 — CHANGES_REQUESTED** [review]

File: uptrain/operators/language/model_grade.py (hunk 5 above)
Comment (sourabhagr) [reply uptrain/operators/language/model_grade.py:108]: No, for model grading, we can fix the temperature. It is defined for response generation only

File: uptrain/framework/checks.py (hunk 6 above)
Comment (sourabhagr) [reply uptrain/framework/checks.py]: Done

File: uptrain/framework/remote.py (hunk 4 above)
Comment (sourabhagr) [reply uptrain/framework/remote.py]: Done

File: uptrain/operators/io/readers.py [hunk 7]
```diff
@@ -9,9 +9,30 @@
...
+    Attributes:
+        fpath (str): Path to the xlsx file.
+        batch_size (Optional[int]): Number of rows to read at a time. Defaults to None, which reads the entire file.
+
+    """
+
+    fpath: str
+    batch_size: t.Optional[int] = None
+
+    def setup(self, settings: Settings):
```
[the remark this answers is not in the document]
Comment (sourabhagr) [reply uptrain/operators/io/readers.py:32]: Done

**Review by ananis25 — APPROVED** [review]

**Pull request merged**